# Data Wrangling

Objetivo: convertir `Outcome` en una etiqueta binaria `Class` y preparar el dataset para EDA y ML.

In [ ]:
import pandas as pd
url='https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBM-DS0321EN-SkillsNetwork/datasets/dataset_part_2.csv'
df=pd.read_csv(url)
print(df.shape)
print(df[['LaunchSite','Orbit','Outcome','Class']].head())

(90, 19)
   LaunchSite Orbit     Outcome  Class
0  CCAFS SLC 40   LEO  None None      0
1  CCAFS SLC 40   LEO  None None      0
2  CCAFS SLC 40   ISS  None None      0
3  VAFB SLC 4E    PO  False Ocean      0
4  CCAFS SLC 40   GTO  None None      0


In [ ]:
print('Lanzamientos por sitio:')
print(df['LaunchSite'].value_counts())
print('Órbitas:')
print(df['Orbit'].value_counts())
print('Success rate:',df['Class'].mean())

Lanzamientos por sitio:
CCAFS SLC 40    55
KSC LC 39A      22
VAFB SLC 4E     13
Name: LaunchSite, dtype: int64
Órbitas:
GTO      27
ISS      21
VLEO     14
PO        9
LEO       7
SSO       5
MEO       3
ES-L1     1
HEO       1
SO        1
GEO       1
Name: Orbit, dtype: int64
Success rate: 0.6666666666666666


### Resultado clave
El conjunto final del curso contiene 90 filas y 19 columnas; `Class=1` representa aterrizaje exitoso y `Class=0` no exitoso. La tasa global de éxito de esta etapa es 66.67%.